developer/build_plugin.py

```diff
@@ -454,23 +537,94 @@ def main():
     """
 
-    usage = "%prog source_path target_path (run with --help for more information)"
+    usage = "%prog source_path [-dskc] target_path (run with --help for more information)"
 
     desc = "Builds a standard toolkit plugin structure ready for testing and deploy"
 
     epilog = """
-Examples:
+
+Details and Examples
+--------------------
+
+In its simplest form, just provide a source and target folder for the build.
 
 > python build_plugin.py ~/dev/tk-maya/plugins/basic /tmp/maya-plugin
 
+For automated build setups, you can provide a specific shotgun API script name and
+and corresponding script key:
+
+> python build_plugin.py
+            --shotgun-host='https://mysite.shotgunstudio.com'
+            --shotgun-script-name='plugin_build'
+            --shotgun-script-key='<script-key-here>'
+            ~/dev/tk-maya/plugins/basic /tmp/maya-plugin
+
+By default, the build script will use the latest app store core for its bootstrapping.
+If you want to use a specific core for the bootstrap, this can be specified via the
+--bootstrap-core-uri option:
+
+> python build_plugin.py
+            --bootstrap-core-uri='sgtk:descriptor:dev?path=~/dev/tk-core'
+            ~/dev/tk-maya/plugins/basic /tmp/maya-plugin
+
+For information about the various descriptors that can be used, see
+http://developer.shotgunsoftware.com/tk-core/descriptor
+
+
 """
     parser = OptionParserLineBreakingEpilog(usage=usage, description=desc, epilog=epilog)
 
-    parser.add_option("-d", "--debug",
-                      default=False,
-                      action="store_true",
-                      help="Enable debug logging"
-                      )
+    parser.add_option(
+        "-d",
+        "--debug",
+        default=False,
+        action="store_true",
+        help="Enable debug logging"
+    )
+
+    parser.add_option(
+        "-c",
+        "--bootstrap-core-uri",
+        default=None,
+        action="store",
+        help=("Specify which version of core to be used by the bootstrap process. "
+            "If not specified, defaults to the most recently released core.")
+    )
+
+    group = optparse.OptionGroup(
+        parser,
+        "Shotgun Authentication",
+        "In order to download content from the Toolkit app store, the script will need to authenticate "
+        "against any shotgun site. By default, it will use the toolkit authentication APIs stored "
+        "credentials, and if such are not found, it will prompt for site, username and password."
+    )
+
+    group.add_option(
+        "-s",
+        "--shotgun-host",
+        default=None,
+        action="store",
+        help="Shotgun Script to use to authenticate to retrieve app store credentials."
+    )
+
+    group.add_option(
+        "-n",
+        "--shotgun-script-name",
+        default=None,
+        action="store",
+        help="Shotgun Script to use to authenticate to retrieve app store credentials."
```

Comment:
Maybe reword this `help` string. Suggestion:

`help="Script name to use for Shotgun authentication"`